Some architectural changes

dlmarion opened the pull request:
Created an interface, TabletHostingServer, that both TabletServer and ScanServer implement.
ScanServer no longer extends TabletServer, which means that we can back out some of the changes
that we made to TabletServer early in feature development. Backed out the constructor changes
to Tablet and created ScanServer.ReadOnlyTablet which overrides methods called in the constructor
that set up the WAL and modify metadata to do nothing.

dlmarion committed:
Removed Thrift service hierarchy in preparation for service changes

This change modifies TabletClientService such that it no longer extends
ClientService and ManagerClientService such that it no longer extends
FateService. This is in preparation for splitting TabletClientService
into at least two services for the ScanServer feature. The TabletServer
and Manager processes needed to be modified to start two Thrift Processors
instead of one. This required the use of the TMultiplexedProcessor object
on the server side, which takes one or more Processors and places them in
a map keyed on the Thrift service name. Clients, when communicating with
the TabletServer or Manager, then need to use a TMultiplexedProtocol with
the name of the Thrift service that they want to call.

dlmarion committed:
wip changes

dlmarion committed:
Separate the tablet scanning portion of the TabletClientService thrift api

dlmarion committed:
Backing out the creation of the TabletScanningServer interface, for now.

dlmarion committed:
Merge branch '2411_external_isolated_scanner' into 2411_with_thrift_changes

dlmarion committed:
Addressed PR comments, consolidated creation of Thrift client and server objects

Conflicts:
	core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java
	core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchReaderIterator.java
	core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftScanner.java
	core/src/main/java/org/apache/accumulo/core/rpc/ThriftUtil.java
	server/base/src/main/java/org/apache/accumulo/server/manager/LiveTServerSet.java
	server/base/src/main/java/org/apache/accumulo/server/util/VerifyTabletAssignments.java
	server/monitor/src/main/java/org/apache/accumulo/monitor/Monitor.java
	server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java
	test/src/main/java/org/apache/accumulo/test/performance/NullTserver.java

dlmarion committed:
Fix dependency issue

dlmarion committed:
Added TabletScanClientService to ThriftClientTypes and ThriftServerTypes

dlmarion committed:
fix formatting

dlmarion committed:
Merge branch '2411_external_isolated_scanner' into 2411_with_thrift_changes

dlmarion committed:
Some architectural changes

Created an interface, TabletHostingServer, that both TabletServer and ScanServer implement.
ScanServer no longer extends TabletServer, which means that we can back out some of the changes
that we made to TabletServer early in feature development. Backed out the constructor changes
to Tablet and created ScanServer.ReadOnlyTablet which overrides methods called in the constructor
that set up the WAL and modify metadata to do nothing.

dlmarion committed:
Fix issue found by errorprone

dlmarion committed:
Revert changes to TabletServer now that ScanServer does not extend TabletServer

dlmarion committed:
Fix errorprone issues

dlmarion committed:
Add missing overrides

keith-turner commented on server/tserver/src/main/java/org/apache/accumulo/tserver/TabletHostingServer.java:
```diff
@@ -0,0 +1,125 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public interface TabletHostingServer {
```
Wondering the following would work and then could drop the getContext() method.

```suggestion
public abstract class TabletHostingServer extends AbstractServer {
```

keith-turner reviewed (commented):
Still looking at this.

dlmarion committed:
Addressed PR comments

Conflicts:
	server/base/src/main/java/org/apache/accumulo/server/rpc/ThriftProcessorTypes.java
	server/tserver/src/main/java/org/apache/accumulo/tserver/TabletClientHandler.java
	server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

dlmarion committed:
Addressed PR comments

Conflicts:
	server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

dlmarion committed:
Merge branch '2411_external_isolated_scanner' into 2411_with_thrift_changes

dlmarion committed:
Merge branch '2411_with_thrift_changes' into 2411_thrift_changes_plus_tablet_changes

Conflicts:
	server/tserver/src/main/java/org/apache/accumulo/tserver/ScanServer.java

dlmarion committed:
Merge branch '2411_external_isolated_scanner' into 2411_with_thrift_changes

Conflicts:
	core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java
	core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchReaderIterator.java
	core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftScanner.java
	core/src/main/java/org/apache/accumulo/core/rpc/ThriftClientTypes.java
	core/src/main/java/org/apache/accumulo/core/rpc/ThriftUtil.java
	core/src/main/thrift-gen-java/org/apache/accumulo/core/tabletserver/thrift/TabletClientService.java
	core/src/main/thrift/tabletserver.thrift
	server/base/src/main/java/org/apache/accumulo/server/manager/LiveTServerSet.java
	server/base/src/main/java/org/apache/accumulo/server/rpc/ThriftProcessorTypes.java
	server/base/src/main/java/org/apache/accumulo/server/util/VerifyTabletAssignments.java
	server/monitor/src/main/java/org/apache/accumulo/monitor/Monitor.java
	server/tserver/src/main/java/org/apache/accumulo/tserver/ScanServer.java
	server/tserver/src/main/java/org/apache/accumulo/tserver/TabletClientHandler.java
	server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java
	server/tserver/src/test/java/org/apache/accumulo/tserver/ScanServerTest.java
	test/src/main/java/org/apache/accumulo/test/performance/NullTserver.java

dlmarion committed:
Merge branch '2411_external_isolated_scanner' into 2411_with_thrift_changes

Conflicts:
	server/tserver/src/main/java/org/apache/accumulo/tserver/ScanServer.java

dlmarion committed:
Share the write tracker between the TabletClientService and the TabletScanClientService

dlmarion committed:
Add override

dlmarion committed:
Merge branch '2411_with_thrift_changes' into 2411_thrift_changes_plus_tablet_changes

Conflicts:
	server/tserver/src/main/java/org/apache/accumulo/tserver/ScanServer.java
	server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java
	server/tserver/src/test/java/org/apache/accumulo/tserver/ScanServerTest.java

dlmarion committed:
remove catch for unthrown Exception

dlmarion committed:
Renamed ThriftClientType for scanning a tablet, moved halt and fastHalt back to TabletClientService

dlmarion committed:
Merge branch '2411_with_thrift_changes' into 2411_thrift_changes_plus_tablet_changes

Conflicts:
	server/tserver/src/main/java/org/apache/accumulo/tserver/ScanServer.java
	server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java

dlmarion committed:
Merge branch '2411_external_isolated_scanner' into 2411_thrift_changes_plus_tablet_changes

Conflicts:
	core/src/main/java/org/apache/accumulo/core/rpc/ThriftClientTypes.java
	core/src/main/thrift-gen-java/org/apache/accumulo/core/tabletserver/thrift/TabletClientService.java
	core/src/main/thrift/tabletserver.thrift
	server/base/src/main/java/org/apache/accumulo/server/rpc/ThriftProcessorTypes.java
	server/tserver/src/main/java/org/apache/accumulo/tserver/ScanServer.java
	server/tserver/src/main/java/org/apache/accumulo/tserver/TabletClientHandler.java
	server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java
	server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftScanClientHandler.java
	server/tserver/src/test/java/org/apache/accumulo/tserver/ScanServerTest.java
	test/src/main/java/org/apache/accumulo/test/performance/NullTserver.java
